bug: fix broken edit page

thelostone-mc opened the pull request:
#### Description 

![image](https://user-images.githubusercontent.com/5358146/133061556-5d5711fe-2c14-4b6c-88b1-33a5ef9d0ec6.png)

mds1 commented:
@thelostone-mc I missed this when we merged the `InputRow.vue` component, but can you or @scco make some small updates to that component:
1. Update the prop definitions there to include `required` and `default` fields. Right now they only define type, so it's not clear which are required. This PR often uses `<InputRow :deleteable="false" :intended="false" text="">`, and it'd be cleaner if all of those unused attributes are just left out, e.g. `text=""` is odd in particular
2. Add comments describing each prop:
    1. `deletable` adds an X icon, but there's no way to attach a method to the X icon, so `deletable` is useless currently and should either be fixed or removed
    2. The purpose of the `intended` prop is not clear
3. Why is `text` a prop but `label` and `input` slots? Seems they should all be slots

mds1 commented on app/src/views/GrantRegistryGrantDetail.vue:
```diff
@@ -105,82 +105,120 @@
-            :rules="isDefined"
-            errorMsg="Please enter a name"
-          />
+          <InputRow :deleteable="false" :intended="false" text="contributions and matching funds are sent to">
```
Current phrasing is missing a word
```suggestion
          <InputRow :deleteable="false" :intended="false" text="address contributions and matching funds are sent to">
```

Or something like "address all received funds are sent to"

mds1 commented on app/src/views/GrantRegistryGrantDetail.vue:
```diff
@@ -105,82 +105,120 @@
 
           <!-- Submit and cancel buttons -->
-          <div class="flex justify-end pt-6">
+          <div class="px-4 md:px-12 py-12">
```
The Cancel and Update Grant buttons are not aligned

![image](https://user-images.githubusercontent.com/17163988/133082597-69a2a237-2c58-4c0c-aa4a-e35676aa1936.png)

thelostone-mc replied:
fixed ! dankee

thelostone-mc commented:
@mds1 


- Added default + Did this and updated codebase to remove redundant code
- Agree with removing deletable if it doesn't work (do we wanna do it here / in a follow up PR)
- @scco - > could you comment on what intent is used for 
- @scco  -> Why is text a prop but label and input slots? Seems they should all be slots
- BaseInput already exists. We should rename it to BaseInputRow ? Although the name might be confusing.

mds1 commented:
> BaseInput already exists. We should rename it to BaseInputRow ? Although the name might be confusing.

Oops, yes I meant to write `BaseInputRow.vue` above. I actually it's less confusing, because you're almost always going to be using ``BaseInputRow.vue` with ` `BaseInput.vue`, which follows the style guide [here](https://v3.vuejs.org/style-guide/#tightly-coupled-component-names-strongly-recommended) and [here](https://v3.vuejs.org/style-guide/#base-component-names-strongly-recommended)

> Agree with removing deletable if it doesn't work (do we wanna do it here / in a follow up PR)

I'm ok with doing that in this PR. Alternatively we can keep the prop _if we're certain we'll need it in the future_ and add a TODO comment explaining it

apbendi commented:
Looking much better @thelostone-mc, thanks for tackling this one. A couple small things I noticed:

* There's no padding at the bottom of the page, so the buttons bump straight into the edge of the browser:

<img width="1225" alt="Screen Shot 2021-09-13 at 9 27 37 AM" src="https://user-images.githubusercontent.com/1479802/133095407-60fbf9bb-0dbc-42a4-9f54-74e11a1c84e1.png">

* In mobile mode, I'm not sure this right hand padding makes sense on the input fields. Wouldn't we want to use the whole space?

<img width="408" alt="Screen Shot 2021-09-13 at 9 27 07 AM" src="https://user-images.githubusercontent.com/1479802/133095573-2bf9c4a7-d888-4bd8-81e9-961c0730acf6.png">

thelostone-mc commented:
@apbendi / @mds1  looks like @scco is fixing em up in another commit .
thoughts on getting this in and fixing it up

gdixon reviewed (approved):
LGTM!
